src/timeMachine/components/view_options/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {FC, useState} from 'react'
+import {useSelector} from 'react-redux'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+
+import {getVisTable} from 'src/timeMachine/selectors'
+
+import {useVisXDomainSettings} from 'src/shared/utils/useVisDomainSettings'
+
+interface Props {
+  setDomain: (xDomain: [number, number]) => void
+  xDomain: number[] | null
+}
+
+const isValidXDomain = xDomain => {
+  return (
+    xDomain &&
+    xDomain.length === 2 &&
+    typeof xDomain[0] === 'number' &&
+    typeof xDomain[1] === 'number'
+  )
+}
+
+export const TimeDomainAutoToggle: FC<Props> = ({setDomain, xDomain}) => {
+  const [isActive, setIsActive] = useState<boolean>(false)
+
+  const {table} = useSelector(getVisTable)
+
+  const [xDom] = useVisXDomainSettings(
```

Comment:
`xDom` is a really basic name. chopping off the end of another variable isn't generally a great strategy for writing readable code. this is the new domain right? why not call it that?